pkg/kubeinteraction/resultsannotation_test.go

```diff
@@ -0,0 +1,78 @@
+package kubeinteraction
+
+import (
+	"encoding/json"
+	"errors"
+	"testing"
+
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/info"
+	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"gotest.tools/v3/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestAddResultsAnnotation(t *testing.T) {
+	testCases := []struct {
+		name          string
+		event         *info.Event
+		expectedError error
+	}{
+		{
+			name: "Valid Event",
+			event: &info.Event{
+				Repository:        "tektoncd/results",
+				SHA:               "8789abb6",
+				EventType:         "PR",
+				PullRequestNumber: 123,
+			},
+			expectedError: nil,
+		},
+		{
+			name:          "Empty Event",
+			event:         &info.Event{},
```

Comment:
if event is empty then accessing attributes of event such as `event.SHA` will fail right :thinking:

Reply:
The fields of the event that is being used for creating `ResultsAnnotation` should have default values in case it's an empty struct, right? I think that's why the unit tests pass.